contracts/network/BancorNetwork.sol

```diff
@@ -1247,7 +1247,8 @@ contract BancorNetwork is IBancorNetwork, Upgradeable, ReentrancyGuardUpgradeabl
                 params.limit
             );
 
-        // if the target token is BNT, notify the BNT pool on collected fees
+        // if the target token is BNT, notify the BNT pool on collected fees (which shouldn't include the network fee
+        // amount, so we have to deduct it explicitly from the full trading fee amount)
```

Comment:
Unrelated, but we can take the opportunity and call this function only if the amount is larger than zero, in order to save the gas cost of an external delegate call (in addition, I'd remove the 'amount == 0' check inside `onFeesCollected`, because it is used there only for gas optimization).
That being said, the whole thing is really worth the effort, if we actually expect this amount to ever be zero.

Reply:
The trading fee is extremely unlikely to be 0. It can only happen if it's set to 0 or the network fee amount is set to 100%.

Reply:
Or the trading amount is really small (such that the trading fee fraction of it becomes zero)...
Resolving.